consensus/src/consensus_observer/network/observer_message.rs

```diff
@@ -361,16 +361,87 @@ impl PayloadWithProofAndLimit {
     }
 }
 
+#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
+pub enum TransactionsWithProof {
+    TransactionsWithProofAndLimits(TransactionsWithProofAndLimits),
+}
+
+impl TransactionsWithProof {
+    pub fn transactions(&self) -> Vec<SignedTransaction> {
+        match self {
+            TransactionsWithProof::TransactionsWithProofAndLimits(payload) => {
+                payload.payload_with_proof.transactions.clone()
+            },
+        }
+    }
+
+    pub fn proofs(&self) -> Vec<ProofOfStore> {
+        match self {
+            TransactionsWithProof::TransactionsWithProofAndLimits(payload) => {
+                payload.payload_with_proof.proofs.clone()
+            },
+        }
+    }
+
+    pub fn transaction_limit(&self) -> Option<u64> {
+        match self {
+            TransactionsWithProof::TransactionsWithProofAndLimits(payload) => {
+                payload.transaction_limit
+            },
+        }
+    }
+
+    pub fn gas_limit(&self) -> Option<u64> {
+        match self {
+            TransactionsWithProof::TransactionsWithProofAndLimits(payload) => payload.gas_limit,
+        }
+    }
+}
+
+/// The transaction payload and proof of each block with a transaction and block gas limit
+#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
+pub struct TransactionsWithProofAndLimits {
+    payload_with_proof: PayloadWithProof,
+    transaction_limit: Option<u64>,
+    gas_limit: Option<u64>,
+}
+
+impl TransactionsWithProofAndLimits {
+    pub fn new(
+        payload_with_proof: PayloadWithProof,
+        transaction_limit: Option<u64>,
+        gas_limit: Option<u64>,
+    ) -> Self {
+        Self {
+            payload_with_proof,
+            transaction_limit,
+            gas_limit,
+        }
+    }
+
+    #[cfg(test)]
+    /// Returns an empty payload with proof and limit (for testing)
+    pub fn empty() -> Self {
+        Self {
+            payload_with_proof: PayloadWithProof::empty(),
+            transaction_limit: None,
+            gas_limit: None,
+        }
+    }
+}
+
 /// The transaction payload of each block
 #[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
 pub enum BlockTransactionPayload {
-    InQuorumStore(PayloadWithProof),
-    InQuorumStoreWithLimit(PayloadWithProofAndLimit),
+    // TODO: deprecate InQuorumStore* variants
+    DeprecatedInQuorumStore(PayloadWithProof),
+    DeprecatedInQuorumStoreWithLimit(PayloadWithProofAndLimit),
     QuorumStoreInlineHybrid(PayloadWithProofAndLimit, Vec<BatchInfo>),
     OptQuorumStore(
-        PayloadWithProofAndLimit,
+        TransactionsWithProof,
```

Comment:
We're okay breaking the existing serialization for the `OptQuorumStore`?

Reply:
^ yeah, this is from @ibalajiarun 
> On second thought, we don't need a V2 for opt quorum store because we haven't turned it on yet?

Reply:
Awesome, thanks!